src/cascadia/Remoting/Monarch.cpp

```diff
@@ -102,17 +100,71 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
     // - the peasant if it exists in our map, otherwise null
     Remoting::IPeasant Monarch::_getPeasant(uint64_t peasantID)
     {
-        auto peasantSearch = _peasants.find(peasantID);
-        return peasantSearch == _peasants.end() ? nullptr : peasantSearch->second;
+        try
+        {
+            auto peasantSearch = _peasants.find(peasantID);
+            auto maybeThePeasant = peasantSearch == _peasants.end() ? nullptr : peasantSearch->second;
+            if (maybeThePeasant)
+            {
+                maybeThePeasant.GetPID();
+            }
+            return maybeThePeasant;
+        }
+        catch (...)

```

Comment:
Why the catch? I see a lot of `GetPID()` that are not surrounded by try-catch.

Reply:
Those should all be fixed now, but go call them out if you see any others